astropy/units/physical.py

```diff
@@ -311,6 +313,19 @@ def __init__(self, unit, physical_types):
     def __iter__(self):
         yield from self._physical_type_list
 
+    def __getattr__(self, attr):
+        if attr in dir(str):
+            warning_message = (
+                f"support for accessing str attributes such as {attr} "
+                "from PhysicalType instances will be removed in a "
+                "subsequent release."
+            )
+            warnings.warn(warning_message, AstropyDeprecationWarning)
+            self_str_attr = getattr(str(self), attr, None)
+            return self_str_attr
+        else:
+            super().__getattribute__(attr)
```

Comment:
Is this here to raise the correct error + message? I don't think that in normal usage this line will ever NOT error.

Reply:
Yes, that's pretty much the purpose.  It gets run to give the right error when doing something like:
```Python
>>> u.Pa.physical_type.asdfhjklasdlk
```
Without it, the above line will return `None`.

Reply:
Alternatively, if you don't care about the message, just explicitly raise the AttributeError. The current form is a little obfuscatory.

Reply:
Huh...I tried raising the `AttributeError`, and got a `RecursionError` instead.  For the time being, I'll add a comment to clarify.

Reply:
That's odd. Here's an example of something similar.

<img width="652" alt="Screen Shot 2021-04-25 at 14 21 49" src="https://user-images.githubusercontent.com/8949649/116004625-9a46a100-a5d1-11eb-8926-de8cb3c46de0.png">

Reply:
Oh...looks like the `RecursionError` came from me using `self.__name__` in the error message, and not `self.__class__.__name__`.  Thank you for checking!